src/main/java/uk/gov/hmcts/reform/coh/controller/RelistingController.java

```diff
@@ -0,0 +1,130 @@
+
+import java.time.Clock;
+import java.util.Date;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.UUID;
+import javax.validation.Valid;
+
+@RestController
+@RequestMapping(
+    value = "/continuous-online-hearings/{onlineHearingId}/relist",
+    produces = MediaType.APPLICATION_JSON_VALUE
+)
+public class RelistingController {
+
+    @Autowired
+    private OnlineHearingService onlineHearingService;
+
+    @Autowired
+    private OnlineHearingStateService onlineHearingStateService;
+
+    @Autowired
+    private SessionEventService sessionEventService;
+
+    @Autowired
+    private Clock clock;
+
+    @ApiOperation(value = "Get re-listing", notes = "A GET request to retrieve a re-listing.")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "Success", response = RelistingResponse.class),
+        @ApiResponse(code = 401, message = "Unauthorised"),
+        @ApiResponse(code = 403, message = "Forbidden"),
+        @ApiResponse(code = 404, message = "Not Found")
+    })
+    @GetMapping
+    public ResponseEntity retrieveRelisting(@PathVariable UUID onlineHearingId) {
+        Optional<OnlineHearing> optionalOnlineHearing = onlineHearingService.retrieveOnlineHearing(onlineHearingId);
+        if (!optionalOnlineHearing.isPresent()) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body("Not found");
+        }
+        OnlineHearing onlineHearing = optionalOnlineHearing.get();
+        RelistingResponse response = new RelistingResponse(
+            onlineHearing.getRelistReason(),
+            onlineHearing.getRelistState(),
+            onlineHearing.getRelistCreated(),
+            onlineHearing.getRelistUpdated()
+        );
+        return ResponseEntity.ok(response);
+    }
+
+    @ApiOperation(value = "Set re-listing reason and state",
+        notes = "A POST request with a request body is used to update re-listing of online hearing.")
+    @ApiResponses(value = {
+        @ApiResponse(code = 202, message = "Accepted"),
+        @ApiResponse(code = 401, message = "Unauthorised"),
+        @ApiResponse(code = 403, message = "Forbidden"),
+        @ApiResponse(code = 404, message = "Not Found"),
+        @ApiResponse(code = 409, message = "Conflict")
+    })
+    @PostMapping
+    public ResponseEntity createDraft(
+        @PathVariable UUID onlineHearingId,
+        @RequestBody @Valid RelistingRequest body
+    ) {
+        Optional<OnlineHearing> optionalOnlineHearing = onlineHearingService.retrieveOnlineHearing(onlineHearingId);
+        if (!optionalOnlineHearing.isPresent()) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body("Not found");
+        }
+
+        if (body.state == RelistingState.ISSUE_PENDING) {
+            return ResponseEntity.badRequest().body("Invalid state");
+        }
+
+        OnlineHearing onlineHearing = optionalOnlineHearing.get();
+
+        Date now = Date.from(clock.instant());
+
```

Comment:
If you had separate POST and PUT, the logic below would be a lot less convoluted .